app/src/js/components/Modal/InactivityModal.js

```diff
@@ -0,0 +1,109 @@
+import React, { useEffect, useState, useCallback } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import get from 'lodash/get';
+import DefaultModal from './modal';
+import { logout } from '../../actions';
+
+const InactivityModal = ({
+  title = 'Session Timeout Warning',
+  children = `We have noticed that you have been inactive for a while.
+  We will close this session in 5 minutes. If you want to stay signed in, select ‘Continue Session’.`,
+  dispatch
+}) => {
+  const [lastKeyPress, setLastKeyPress] = useState(Date.now());
+  const [hasModal, setHasModal] = useState(false);
+
+  function handleConfirm() {
+    setLastKeyPress(Date.now());
+    closeModal();
+  }
+
+  const handleLogout = useCallback(() => {
+    dispatch(logout()).then(() => {
+      if (get(window, 'location.reload')) {
+        window.location.reload();
+      }
+    });
+  }, [dispatch]);
+
+  function closeModal() { // the X botton
+    setLastKeyPress(Date.now());
+    if (hasModal) {
+      setHasModal(false); // hide modal if user resumes activity
+    }
+  }
+
+  // Effect to setup event listeners for keyboard activity
+  useEffect(() => {
+    // Function to update the lastKeyPress time
+    const handleKeypress = () => {
+      setLastKeyPress(Date.now());
+      if (hasModal) {
+        setHasModal(false); // hide modal if user resumes activity
+      }
+    };
+    window.addEventListener('keydown', handleKeypress);
+
+    return () => {
+      window.removeEventListener('keydown', handleKeypress);
+    };
+  }, [hasModal]);
+
+  // Effect to handle showing the modal after 30 minutes of inactivity
+  useEffect(() => {
+    const checkInactivity = setInterval(() => {
+      if (Date.now() - lastKeyPress > 1800000 && !hasModal) { // 1800000 ms = 30 minutes
```

Comment:
Can we make the inactivity time configurable?

Reply:
What's the difference between this inactivity timeout and session timeout?
The launchpad session can timeout regardless this inactivity, is it correct?  Does the launchpad Session get refresh when we click 'Continue Session'?

Reply:
**Can you add a cypress test to verify the functionality?**
I don't think this is necessary as the other modals do not have tests written for this so it shouldn't be a requirement. If it is necessary I would be happy to do it but will need to be shown how as I normally do not deal with Cypress. 

**The Modal says 'We will close this session in 5 minutes. ', but after 5 minutes, the Session is not closed or log me out, and the modal window is still there. I click 'Continue Session', the modal window closes and the session is still there.**
In progress

**Can we make the inactivity time configurable?**
Done

**What's the difference between this inactivity timeout and session timeout? How do they work together?
The launchpad session can timeout regardless this inactivity, is it correct? Does the launchpad Session get refresh when we click 'Continue Session'?**
Currently the session timeout just keeps track if the user is logged in or not. I am not sure what this timeout is set for. The inactivity modal will keep track of the users activity. If there is no activity after 30 min then a pop up will show and a second timer will start. This second timer is set for 5 min. If the user wishes to continue the session then they will need to interact with the modal and select "Continue session" otherwise the user will be logged out. In the future the session timeout and the inactivity timeout will be merged but this will be addressed in another ticket.